src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -565,16 +566,54 @@ def rshift(self, y: int) -> "ReplicatedSharedTensor":
 
         Raises:
             ValueError: If y is not an integer.
+            ValueError : If invalid shift value is provided.
         """
         if not isinstance(y, int):
             raise ValueError("Right Shift works only with integers!")
 
+        ring_bits = get_nr_bits(self.ring_size)
+        if y > ring_bits - 1:
+            raise ValueError(
+                f"Invalid value for right shift : {y},must be in range:[0,{ring_bits-1}]"
+            )
+
         res = ReplicatedSharedTensor(
             session_uuid=self.session_uuid, config=self.config, ring_size=self.ring_size
         )
         res.shares = [share >> y for share in self.shares]
         return res
 
+    def bit_extraction(self, pos: int = 0) -> "ReplicatedSharedTensor":
+        """Extracts the bit at the specified position.
+
+        Args:
+            pos (int): position to extract bit.
+
+        Returns:
+            ReplicatedSharedTensor : extracted bits at specific position.
+
+        Raises:
+            ValueError: If invalid position is provided.
+        """
+        ring_bits = get_nr_bits(self.ring_size)
+        if pos > ring_bits - 1:
+            raise ValueError(
+                f"Invalid position for bit_extraction : {pos},must be in range:[0,{ring_bits-1}]"
+            )
+        shares = []
+        # logical shift
+        bit_mask = torch.ones(self.shares[0].shape, dtype=self.shares[0].dtype) << pos
+        for share in self.shares:
+            tensor = share & bit_mask
+            shares.append(tensor)
+        rst = ReplicatedSharedTensor(
+            shares=shares,
+            session_uuid=self.session_uuid,
+            config=Config(encoder_base=1, encoder_precision=0),
```

Comment:
Q: Can't we only set the encoder_precision?

Reply:
Yeah that would also work , I would change it.

Reply:
No no...it is also fine like this :D since this is how we use the Config in all the file :smile:  (I was asking more if there was a reason for that)

Reply:
Oh ok ,no specific reason george